datasets/interpress_news_category_tr/interpress_news_category_tr.py

```diff
@@ -0,0 +1,115 @@
+# coding=utf-8
+# Lint as: python3
+"""Turkish News Category Dataset (270K) - Interpress Media Monitoring Company"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import logging
+import os
+
+import datasets
+
+
+_DESCRIPTION = """\
+It is a Turkish news data set consisting of 273601 news in 17 categories, compiled from print media and news websites between 2010 and 2017 by the Interpress (https://www.interpress.com/) media monitoring company.
+"""
+
+_CITATION = ""
+
+_HOMEPAGE = "https://www.interpress.com/"
+_DOWNLOAD_URL = "https://www.interpress.com/downloads/interpress_news_category_tr_270k.zip"
+_DATASET_URLS = {
+    "train": "interpress_news_category_tr_270k_train.tsv",
+    "test": "interpress_news_category_tr_270k_test.tsv",
+}
+
+
+class InterpressNewsCategoryTRConfig(datasets.BuilderConfig):
+    """BuilderConfig for InterpressNewsCategoryTR."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for InterpressNewsCategoryTR.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(InterpressNewsCategoryTRConfig, self).__init__(**kwargs)
+
+
+class InterpressNewsCategoryTR(datasets.GeneratorBasedBuilder):
+    """Turkish News Category Dataset (270K) - Interpress Media Monitoring Company"""
+
+    BUILDER_CONFIGS = [
+        InterpressNewsCategoryTRConfig(
+            name="270k",
+            version=datasets.Version("1.0.0"),
+            description="Turkish News Category Dataset (270K) - Interpress Media Monitoring Company",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "title": datasets.Value("string"),
+                    "content": datasets.Value("string"),
+                    "category": datasets.features.ClassLabel(
+                        names=[
+                            "aktuel",
+                            "bilisim",
+                            "egitim",
+                            "ekonomi",
+                            "gida",
+                            "iletisim",
+                            "kultursanat",
+                            "magazin",
+                            "saglik",
+                            "savunma",
+                            "seyahat",
+                            "siyasi",
+                            "spor",
+                            "teknoloji",
+                            "ticaret",
+                            "turizm",
+                            "yasam",
+                        ]
+                    ),
+                    "categorycode": datasets.features.ClassLabel(
+                        names=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
```

Comment:
If there are no actual names to provide here, you should use the `num_classes` parameter here.
```suggestion
                        num_classes=17
```
See [the docs](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.ClassLabel)

Reply:
Thanks for editing.